platform/platform_freebsd.cpp

```diff
@@ -0,0 +1,142 @@
+#include "platform/platform.hpp"
```

Comment:
:+1:, virtual base class would be much easier to handle. Now, as all stuff is initialized in ctor, you can't basically use virtual methods, so you have to parametrize base class ctor with e.g. executable path, which is non-flexible.